### PR TITLE
Improved button config helper Quality of Life.

### PR DESCRIPTION
**Pull request by RogPodge**
## Overview
Fixes several issues where the button config helper was not clear with whether the materials on objects were derived from the current icon sets. In addition, added warnings for when certain kinds of iconsets were not loaded, and fixed issues where the icon sets would not load properly in the editor, while also adding the option for users to automatically refresh the sprite icons view

diff1
diff2
warningscompiled
DisappearSpriteTextureFix

## Changes
- Fixes: #9142, #9061

### CODE REVIEW

**Comment by MaxWang-MS** [comment]
Just a minor comment: in the gif above it seems that after switching the icon style from quad to sprite one needs to manually click the refresh button to let the sprite show up. Would it be possible to make that automatic?

**Comment by RogPodge** [comment]
> 
> 
> Just a minor comment: in the gif above it seems that after switching the icon style from quad to sprite one needs to manually click the refresh button to let the sprite show up. Would it be possible to make that automatic?

Not without tanking performance :( Currently the sprite cache is set to reset when the SpriteIcon variable is changed inside the IconSet, however, sprites in the sprite icon can have their import settings updated without touching the encompassing Iconset. Doing any checks during the main editor update loop made Unity pretty chuggy and the experience unpleasant.

File: Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonIconSet.cs [hunk 1]
```diff
@@ -328,15 +330,39 @@ public bool EditorDrawSpriteIconSelector(Sprite currentSprite, out Sprite newSpr
...
+                    {
+                        newSprite = spriteIcons[newSelection];
+                    }
                 }
             }
 
             return newSprite != null;
         }
 
+        public void UpdateSpriteIconTextures()
```
Comment (CDiaz-MS) [inline Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonIconSet.cs:346]: Missing doc comment

**Review by CDiaz-MS — APPROVED** [review]
Overall looks good to me, it would be great if @grayclee could also test the changes.

**Comment by MaxWang-MS** [comment]
> > Just a minor comment: in the gif above it seems that after switching the icon style from quad to sprite one needs to manually click the refresh button to let the sprite show up. Would it be possible to make that automatic?
> 
> Not without tanking performance :( Currently the sprite cache is set to reset when the SpriteIcon variable is changed inside the IconSet, however, sprites in the sprite icon can have their import settings updated without touching the encompassing Iconset. Doing any checks during the main editor update loop made Unity pretty chuggy and the experience unpleasant.

I see... Thanks for looking into the matter!

**Review by MaxWang-MS — APPROVED** [review]

**Pull request merged**